catwalk/models/huggingface.py

```diff
@@ -28,66 +30,71 @@ def predict(  # type: ignore
         *,
         batch_size: int = 32
     ) -> Iterator[Dict[str, Any]]:
+        device = resolve_device()
+
         if task.has_instance_conversion(InstanceFormat.HF_MC):
-            return self._predict_mc(task, instances, batch_size=batch_size)
+            mc_instances = cast(Sequence[HFMCInstance], self._convert_instances(instances, InstanceFormat.HF_MC, task))
+            model = cached_transformers.get(AutoModelForMultipleChoice, self.pretrained_model_name_or_path, False).to(device)
+            tokenizer = cached_transformers.get_tokenizer(AutoTokenizer, self.pretrained_model_name_or_path)
+            return self._predict_mc(mc_instances, model, tokenizer, batch_size=batch_size)
         elif task.has_instance_conversion(InstanceFormat.HF_QA):
-            return self._predict_qa(task, instances, batch_size=batch_size)
+            qa_instances = cast(Sequence[HFQAInstance], self._convert_instances(instances, InstanceFormat.HF_QA, task))
```

Comment:
It's because I need the `_predict_mc()` and `_predict_qa()` methods to take model and tokenizer, so I can call them from `TrainableHFAutoModel`. I didn't want to make another intermediate function just for three lines of code.